src/GNDStk/Node/test/all.test.cpp

```diff
@@ -1,26 +1,96 @@
 
 #include "catch.hpp"
 #include "GNDStk.hpp"
+#include <cstring>
 
 using namespace njoy::GNDStk;
 using namespace mixed::child;
+using namespace mixed::meta;
 
 SCENARIO("Testing GNDStk Node all()") {
 
    GIVEN("A tree read from a particular example GNDS file") {
       // tree
       const Tree<> tree("n-008_O_016.xml");
 
-      // node: chemicalElements
-      const auto &elems = tree(reactionSuite,PoPs,chemicalElements);
+      // chemicalElements / chemicalElement
+      WHEN("We extract all <chemicalElement> nodes within <chemicalElements>") {
+         const auto &CEs = tree(reactionSuite,PoPs,chemicalElements);
+         {
+            // without "found"
+            auto ce = CEs.all("chemicalElement");
+            THEN("We find eight of them") {
+               CHECK(ce.size() == 8);
+            }
+         } {
+            // with "found"
+            bool found = false;
+            auto ce = CEs.all("chemicalElement",found);
+            THEN("We find eight of them") {
+               CHECK(found);
+               CHECK(ce.size() == 8);
+            }
+         }
+      }
+
+      // reactions / reaction
+      WHEN("We extract selected <reactions> nodes within <reaction>") {
+         const auto &Rs = tree(reactionSuite,reactions);
+
+         // filter that requires: node has metadata label="n + *"
+         auto nplus = [](const node &n)
+            { return 0 == strncmp(n(label).c_str(), "n + ", 4); };
+
+         // filter that requires: node has metadata ENDF_MT="600"
+         auto endf_mt_600 = [](const node &n)
+            { return n(ENDF_MT) == 600; };
+
+         // filter that requires: both of the above
+         auto filter_both = [nplus,endf_mt_600](const node &n)
+            { return nplus(n) && endf_mt_600(n); };
+
+         THEN("There are 15 <reaction> nodes with label=\"n + *\"") {
+            CHECK(Rs.all("reaction",nplus).size() == 15);
+         }
+
+         THEN("There is one <reaction> node with ENDF_MT=\"600\"") {
+            CHECK(Rs.all("reaction",endf_mt_600).size() == 1);
+         }
+
+         THEN("There are no <reaction> nodes with both "
+              "label=\"n + *\" and ENDF_MT=\"600\"") {
+            bool found = true;
+            CHECK(Rs.all("reaction",filter_both,found).size() == 0);
+            CHECK(!found);
+         }
+      }
+
+      // reactions / reaction
+      WHEN("We invoke node::all() with an empty (\"\") key") {
+         auto Rs = tree(reactionSuite,reactions);
+
+         THEN("We get back a container with one node, identical to *this") {
```

Comment:
What is the use case for this? It seems like we are getting a copy of a node. Why not just use the copy constructor?

Reply:
This is basically a mechanical test of a certain behavior that doesn't appear at first glance to be useful. However, a certain other behavior - one that's quite useful - is built on top of this. I'll have to write a much longer explanation in order to establish why I'm doing this. At the moment I have some other things to do, and will write the long explanation when I get the chance. However, rest assured that we want this.